src/pseudopeople/configuration.py

```diff
@@ -0,0 +1,203 @@
+from pathlib import Path
+from typing import Dict, NamedTuple, Union
+
+import yaml
+from vivarium.config_tree import ConfigTree
+
+from pseudopeople.schema_entities import FORMS
+
+
+class Keys(NamedTuple):
+    """NamedTuple containing all key names used in the configuration file
+    NOTE: 'additional_parameters' is actually a dict with its own key-values defined
+    """
+
+    ROW_NOISE = "row_noise"
+    OMISSION = "omission"
+    DUPLICATION = "duplication"
+    COLUMN_NOISE = "column_noise"
+    ROW_NOISE_LEVEL = "row_noise_level"
+    TOKEN_NOISE_LEVEL = "token_noise_level"
+    ADDITIONAL_PARAMETERS = "additional_parameters"
+
+
+# Define non-baseline default items
+# NOTE: default values are defined in entity_types.RowNoiseType and entity_types.ColumnNoiseType
+DEFAULT_NOISE_VALUES = {
+    # TODO: Uncomment when omission gets implemented
+    # FORMS.CENSUS.name: {
+    #     Keys.ROW_NOISE: {
+    #         Keys.OMISSION: 0.0145,
+    #     },
+    # },
+    # FORMS.ACS.name: {
+    #     Keys.ROW_NOISE: {
+    #         Keys.OMISSION: 0.0145,
+    #     },
+    # },
+    # FORMS.CPS.name: {
+    #     Keys.ROW_NOISE: {
+    #         Keys.OMISSION: 0.2905,
+    #     },
+    # },
+}
+
+
+def get_configuration(user_configuration: Union[Path, str, Dict] = None) -> ConfigTree:
+    """
+    Gets a noising configuration ConfigTree, optionally overridden by a user-provided YAML.
+
+    :param user_configuration: A path to the YAML file or a dictionary defining user overrides for the defaults
+    :return: a ConfigTree object of the noising configuration
+    """
+
+    default_config_layers = [
+        "baseline",
+        "default",
+        "user",
+    ]
+    noising_configuration = ConfigTree(
+        layers=default_config_layers,
+    )
+
+    # Instantiate the configuration file with baseline values
+    baseline_dict = {}
+    for form in FORMS:
+        if not form.is_implemented:
+            continue
+        baseline_dict[form.name] = {}
```

Comment:
Create the dictionaries as stand-alone dictionaries and only add them to the layer above once the dictionary is complete if they are not empty. This will allow you to remove those `del` calls below.

```
baseline_dict = {}
for form in FORMS:
	form_dict = {}
	row_noise_dict = {}
	column_dict = {}

	for row_noise in for.row_noise_types:
		row_noise_type_dict = {}
		if row_noise.noise_level is not None:
			row_noise_type_dict["noise_level"] = row_noise.noise_level
		if row_noise_type_dict:
			row_noise_dict[row_noise.name] = row_noise_type_dict
	
	for column in forms.columns:
	    column_noise_dict = {}
	    for noise_type in column.noise_types:
			column_noise_type_dict = {}
			if noise_type.row_noise_level is not None:
			    column_noise_type_dict[Keys.ROW_NOISE_LEVEL] = noise_type.row_noise_level
			...
			if column_noise_type_dict:
			    column_noise_dict[noise_type.name] = column_noise_type_dict
		if column_noise_dict:
			column_dict[column.name] = column_noise_dict
		
	if row_noise_dict:
		form_dict[Keys.ROW_NOISE] = row_noise_dict
	if column_dict:
		form_dict[Keys.COLUMN_NOISE] = column_dict
	if form_dict:
	    baseline_dict[form.name] = form_dict
```

Reply:
Wow, I could NOT figure this out yesterday!